src/python/pants/backend/native/tasks/native_external_library_fetch.py

```diff
@@ -154,15 +181,22 @@ def _collect_external_libs(self, results_dir):
     include_dir = os.path.join(results_dir, 'include')
 
     lib_names = []
+    static_archive_paths = []
     if os.path.isdir(lib_dir):
       for filename in os.listdir(lib_dir):
-        lib_name = self._parse_lib_name_from_library_filename(filename)
-        if lib_name:
-          lib_names.append(lib_name)
-
-    return NativeExternalLibraryFiles(include_dir=include_dir,
-                                      lib_dir=lib_dir,
-                                      lib_names=tuple(lib_names))
+        if filename.endswith('.a'):
+          archive_path = os.path.join(lib_dir, filename)
+          static_archive_paths.append(archive_path)
+        else:
+          lib_name = self._parse_lib_name_from_library_filename(filename)
+          if lib_name:
+            lib_names.append(lib_name)
+
+    return NativeExternalLibraryFiles(
+      include_dir=include_dir,
+      lib_dir=lib_dir,
+      lib_names=tuple(lib_names),
+      static_archive_paths=tuple(static_archive_paths))
```

Comment:
We cast as tuple because we want these kwargs to be immutable, correct?

Reply:
Tuples are hashable -- while this particular datatype doesn't go through the v2 engine, it makes it easier if you can assume your input is a tuple for other datatypes which do get produced or consumed by rules.

Reply:
Hm. Immutability is a much better reason than the one I just gave.